server/routes/api/comments/comments.ts

```diff
@@ -360,4 +360,119 @@ router.post(
   }
 );
 
+router.post(
+  "comments.add_reaction",
+  rateLimiter(RateLimiterStrategy.TwentyFivePerMinute),
+  auth(),
+  feature(TeamPreference.Commenting),
+  validate(T.CommentsReactionSchema),
+  transaction(),
+  async (ctx: APIContext<T.CommentsReactionReq>) => {
+    const { id, emoji } = ctx.input.body;
+    const { user } = ctx.state.auth;
+    const { transaction } = ctx.state;
+
+    const comment = await Comment.findByPk(id, {
+      transaction,
+      rejectOnEmpty: true,
+      lock: {
+        level: transaction.LOCK.UPDATE,
+        of: Comment,
+      },
+    });
+    authorize(user, "addReaction", comment);
+
+    await Reaction.findOrCreate({
+      where: {
+        emoji,
+        userId: user.id,
+        commentId: id,
+      },
+      transaction,
+    });
+    const added = await comment.updateReactions({
```

Comment:
Move and split this logic in `updateReactions` into `AfterCreate` and `AfterDestroy` hooks on the `Reaction` model, I promise everything will end up cleaner.

You can then also use the return property from `findOrCreate` to know if it was added or not.

Reply:
It sure did, thank you!